spec/requests/api/providers_spec.rb

```diff
@@ -130,6 +146,24 @@
       expect(response).to have_http_status(:forbidden)
     end
 
+    it "creates provider with custom attributes" do
+      api_basic_authorize collection_action_identifier(:providers, :create)
+      run_post(providers_url, sample_rhevm.merge!("custom_attributes" => custom_attributes))
```

Comment:
I think I'm a bit confused about this, and may have led you in a slightly wrong direction. Is there a reason why custom attributes are not added as a subcollection of providers in the config/api.yml? That will give you some already built in support:

```
POST /api/providers/<provider id>/custom_attributes

{ "action": "add", etc...}

POST /api/providers/<provider id>/custom_attributes/<custom attribute id>

{"action": "edit" ....}
```

And so on